client/app/telecom/pack/xdsl/modem/templates/config/pack-xdsl-modem-templates-config.controller.js

```diff
@@ -0,0 +1,261 @@
+angular.module('managerApp').controller('XdslModemTemplateConfigCtrl', class XdslModemTemplateConfigCtrl {
+  /* @ngInject */
+  constructor($q, $stateParams, $translate, $uibModal, OvhApiXdsl, TucToast, TucToastError) {
+    this.$q = $q;
+    this.$stateParams = $stateParams;
+    this.$translate = $translate;
+    this.$uibModal = $uibModal;
+    this.OvhApiXdsl = OvhApiXdsl;
+    this.TucToast = TucToast;
+    this.TucToastError = TucToastError;
+    this.errors = {
+      template: false,
+    };
+  }
+
+  fetchTemplates() {
+    this.loaders.template = true;
+    this.OvhApiXdsl.TemplateModem().v6().resetAllCache();
+    return this.OvhApiXdsl.TemplateModem().v6().query()
+      .$promise
+      .then(names => this.$q
+        .all(_.map(
+          _.chunk(names, 50),
+          chunkNames => this.OvhApiXdsl.TemplateModem().v6().getBatch({
+            name: chunkNames,
+          }).$promise,
+        ))
+        .then((chunkResult) => {
+          this.loaders.template = false;
+          const templates = [];
+          angular.forEach(_.flatten(chunkResult), (resource) => {
```

Comment:
Since you are using `angular.forEach` and always pushing into your `templates` array it means that it could be turned into a `.map` instead.